### ISSUE TITLE
Feature : configure the migration dir in diesel.toml

### ISSUE THREAD

**Issue by theredfish** [diesel-rs/diesel#2163]
Hello,

Could we use the `diesel.toml` file to configure the migrations directory path? It could be easier than using the `--migration-dir` argument. It seems to be the right place for this kind of configuration. 

If you agree I can provide a pull request for this feature.

**Comment by sgrif** [comment]
Sounds reasonable to me

**Comment by theredfish** [comment]
Hello,
I'm trying to run tests, but I'm getting some errors about duplicate code. It seems that macros for features are ignored. For example : 

```rust
#[cfg(feature = "sqlite")]
const BACKEND: &str = "sqlite";
#[cfg(feature = "postgres")]
const BACKEND: &str = "postgres";
#[cfg(feature = "mysql")]
const BACKEND: &str = "mysql";
```

```bash
    |
119 | const BACKEND: &str = "sqlite";
    | ------------------------------- previous definition of the value `BACKEND` here
120 | #[cfg(feature = "postgres")]
121 | const BACKEND: &str = "postgres";
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `BACKEND` redefined here
    |
    = note: `BACKEND` must be defined only once in the value namespace of this module
```

This is what the compiler returns for the command `cargo test` in `diesel_cli`. I didn't found any specific information in the `CONTRIBUTING.md` about the way to run the tests. Could you help me with this @sgrif ?

**Comment by sgrif** [comment]
The tests for the CLI need to be run for a specific backend. Add `--no-default-features --features some_backend` or look at `bin/test`

**Comment by theredfish** [comment]
@sgrif  I tried to put the logic inside the diesel_cli crate, but I'm not sure if it's the right place. In `diesel_cli/main.rs` the function `migrations_dir` calls `migrations::find_migrations_directory`. 

Should I manage to read the `diesel.toml` file in the `diesel_migrations` crate? It's where I found the function `find_migrations_directory` (so in my case I need to add the feature to load the path from the `diesel.toml` file). 

Thanks

**Comment by ralpha** [comment]
It seems like this is (still) not merges with [diesel_cli](https://crates.io/crates/diesel_cli) 
As I get:
```
unknown field `migrations_directory`, expected `print_schema`
```
And current version is: 1.4.0 From Jan 21, 2019. This issue was created after that date.

Just saying that this would be a nice feature to have.

**Comment by weiznich** [comment]
As no release contains the the corresponding fix this is expected behavior. As this is already fixed on master the next non-patch release will contain the feature.

Closed because already fixed with #2179
